src/images/buffer.rs

```diff
@@ -1406,6 +1443,136 @@ where
+                    // One of them is responsible.
+                    UnsupportedErrorKind::ColorspaceCicp(if from_color.qualify_stability() {
+                        into_color
+                    } else {
+                        from_color
+                    }),
+                ),
+            ));
+        };
+
+        let transform = transform
+            .supported_transform_fn::<FromType, IntoType>()
+            .map_err(ImageError::Unsupported)?;
+
+        Ok(transform)
+    }
+}
+
+impl<C, SelfPixel: Pixel> ImageBuffer<SelfPixel, C>
+where
+    SelfPixel: PixelWithColorType,
+    C: Deref<Target = [SelfPixel::Subpixel]> + DerefMut,
+{
+    /// Copy pixel data from one buffer to another, calculating equivalent color representations
+    /// for the target's color space.
+    ///
+    /// Returns `Ok` if:
+    /// - Both images to have the same dimensions, otherwise returns a [`ImageError::Parameter`].
+    /// - The primaries and transfer functions of both image's color spaces must be supported,
+    ///   otherwise returns a [`ImageError::Unsupported`].
+    /// - The pixel's channel layout must be supported for conversion, otherwise returns a
+    ///   [`ImageError::Unsupported`]. You can rely on RGB and RGBA always being supported. If a
+    ///   layout is supported for one color space it is supported for all of them.
+    ///
+    /// To copy color data of arbitrary channel layouts use `DynamicImage` with the overhead of
+    /// having data converted into and from RGB representation.
+    pub fn copy_from_color<FromType, D>(
+        &mut self,
+        from: &ImageBuffer<FromType, D>,
+        mut options: ConvertColorOptions,
+    ) -> ImageResult<()>
+    where
+        FromType: Pixel<Subpixel = SelfPixel::Subpixel> + PixelWithColorType,
+        D: Deref<Target = [SelfPixel::Subpixel]>,
+    {
+        if self.dimensions() != from.dimensions() {
+            return Err(ImageError::Parameter(ParameterError::from_kind(
+                ParameterErrorKind::DimensionMismatch,
+            )));
+        }
+
+        let transform =
+            options.as_transform::<FromType, SelfPixel>(from.color_space(), self.color_space())?;
+
+        let from = from.inner_pixels();
+        let into = self.inner_pixels_mut();
+
+        debug_assert_eq!(
+            from.len() / usize::from(FromType::CHANNEL_COUNT),
+            into.len() / usize::from(SelfPixel::CHANNEL_COUNT),
+            "Diverging pixel count despite same size",
+        );
+
+        transform(from, into);
+
+        Ok(())
+    }
+
+    /// Convert this buffer into a newly allocated buffer, changing the color representation.
+    ///
+    /// This will avoid an allocation if the target layout or the color conversion is not supported
+    /// (yet).
+    ///
+    /// See [`ImageBuffer::copy_from_color`] if you intend to assign to an existing buffer,
+    /// swapping the argument with `self`.
+    pub fn to_color<IntoType>(
```

Comment:
Perhaps `apply_color_space` or `apply_color_space_transform`? That would parallel the `apply_orientation` method.